api/envoy/config/core/v3/health_check.proto

```diff
@@ -103,8 +103,18 @@ message HealthCheck {
     // [#not-implemented-hide:] HTTP specific payload.
     Payload send = 3;
 
-    // [#not-implemented-hide:] HTTP specific response.
-    Payload receive = 4;
+    // Specifies a list of HTTP expected responses to match in the first ``response_buffer_size`` bytes of the response body.
+    // If it is set, both the expected response check and status code determine the health check.
+    // When checking the response, “fuzzy” matching is performed such that each payload block must be found,
+    // and in the order specified, but not necessarily contiguous.
+    //
+    // .. note::
+    //
+    //   Only the first 1024 bytes of the response body is used to match when ``response_buffer_size`` is not set.
+    repeated Payload receive = 4;
+
+    // Specifies the size of response buffer in bytes that is used to Payload match. Default value is 1024.
+    uint64 response_buffer_size = 14 [(validate.rules).uint64 = {gte: 0}];
```

Comment:
Per [API style-guide](https://github.com/envoyproxy/envoy/blob/main/api/STYLE.md), this should be a well-known-type (e.g., `google.protobuf.UInt32Value`).

Reply:
I removed the default value. But in case future, changed it to the wrapper.